scene/gui/text_edit.cpp

```diff
@@ -4084,7 +4084,7 @@ void TextEdit::cut() {
 		backspace_at_cursor();
 		update();
 		cursor_set_line(cursor.line + 1);
-		cut_copy_line = true;
+		cut_copy_line = clipboard;
```

Comment:
Might be more explicit to keep it a bool and do `cut_copy_line = !clipboard.empty();` instead of storing a string no?

Reply:
No, the reason it's a `String` is to ask whether it's the same clip in order to differentiate between:

(1) copy line in Godot
(2) paste that line in Godot

and

(1) copy line in Godot
(2) copy something else into clipboard in another app
(3) paste that other context into Godot

In the latter case, (3) needs information apart from `cut_copy_line == true` that the current clipboard's content is not actually from (1) but from (2) and thus `cut_copy_line` does not apply.

Other ways to fix this would be:

(1) Add a magic prefix into the saved clipboard text, e.g. `#GODOT_LINE_CUT#:` and check for that when we paste (i.e. if it exists, it replaces the current `cut_copy_line == true` case
(2) Save additional meta information with the stored clipboard data (OS dependent) that tells us it's our clip and it's a "line clip"

Reply:
Ah indeed, thanks for the detailed answer :)